### PR TITLE
[1.1.x] Fix Citadel using in-memory cert without writing to secret issue.

### PR DESCRIPTION
**Pull request by myidpt**
Fix on 1.1 for https://github.com/istio/istio/issues/14512.
Integration test done by: https://github.com/istio/istio/pull/14513

### CODE REVIEW

**Comment by duderino** [comment]
prow/e2e-mixer-no_auth.sh is a known failure unrelated to this change

File: security/pkg/pki/ca/ca.go [hunk 1]
```diff
@@ -148,17 +148,20 @@ func NewSelfSignedIstioCAOptions(ctx context.Context, caCertTTL, certTTL, maxCer
 			return nil, fmt.Errorf("failed to create CA KeyCertBundle (%v)", err)
 		}
 
-		// Rewrite the key/cert back to secret so they will be persistent when CA restarts.
+		// Write the key/cert back to secret so they will be persistent when CA restarts.
 		secret := BuildSecret("", CASecret, namespace, nil, nil, nil, pemCert, pemKey, istioCASecretType)
 		if _, err = client.Secrets(namespace).Create(secret); err != nil {
-			log.Errorf("Failed to write secret to CA (error: %s). This CA will not persist when restart.", err)
+			log.Errorf("Failed to write secret to CA (error: %s). Abort.", err)
+			return nil, fmt.Errorf("failed to create CA due to secret write error")
```
Comment (duderino) [inline security/pkg/pki/ca/ca.go]: What will happen here?  The Citadel process will exit?

Comment (myidpt) [reply security/pkg/pki/ca/ca.go]: That's correct. The process will exit if the CA-root secret already exists and Citadel tries to overwrite it. In this case, the Citadel will be restarted by Kubelet, to read the CA-root secret again.

I think this is good and safe enough for this fix. We may implement retries in the future if necessary.

**Comment by wenchenglu** [comment]
I agree this is the safest change for now while fixing this issue.

Oliver, could you file a github issue for feature improvement to add retry on network timeout? and assign an owner and make sure we keep track of that

**Pull request merged**